packages/utils/src/request.js

```diff
@@ -80,14 +89,14 @@ export const verifyResponseStatus = async response => {
       throw new Error(`Network error ${response.status}`);
     }
     if (
-      responseJson.status &&
-      (responseJson.status < 200 || responseJson.status >= 300) &&
+      response.status &&
+      (response.status < 200 || response.status >= 300) &&
       !responseJson.error
     ) {
-      throw new Error(responseJson.message);
+      throwCustomError(response.status, responseJson.message);
     }
     if (responseJson.error) {
-      throw new Error(responseJson.error);
+      throwCustomError(response.status, responseJson.error);
```

Comment:
This is for resolving something that I promised Edwin I would resolve later. Context is all here: https://github.com/beyondessential/tupaia/pull/1716#discussion_r532389822